crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -0,0 +1,186 @@
+pub mod common;
+
+mod advancing_time_tests {
+
+    use std::time::SystemTime;
+    use std::{thread, time};
+
+    use hyper::Body;
+    use serde_json::json;
+
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::get_json_body;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    #[tokio::test]
+    async fn set_time_in_past() {
+        // set time and assert
+        let past_time = 1;
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let set_time_body = Body::from(
+            json!({
+                "time": past_time
+            })
+            .to_string(),
+        );
+        let resp_set_time = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body_set_time = get_json_body(resp_set_time).await;
+        assert_eq!(resp_body_set_time["block_timestamp"], past_time);
+
+        // create block and check if block_timestamp is greater than past_time
+        let resp_create_block = devnet
+            .post_json("/create_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_create_block = get_json_body(resp_create_block).await;
+        assert!(resp_body_create_block["block_timestamp"].as_u64() > Some(past_time));
+
+        // wait 1 second
+        thread::sleep(time::Duration::from_secs(1));
+
+        // check if after mint block_timestamp is greater than last block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let resp_latest_block = devnet
+            .post_json("/get_latest_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_latest_block = get_json_body(resp_latest_block).await;
+        assert!(
+            resp_body_latest_block["block_timestamp"].as_u64()
+                > resp_body_create_block["block_timestamp"].as_u64()
+        );
+    }
+
+    #[tokio::test]
+    async fn set_time_in_far_future() {
+        // set time and assert
+        let future_time: u64 = 3376684800;
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let set_time_body = Body::from(
+            json!({
+                "time": future_time
+            })
+            .to_string(),
+        );
+        let resp = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body = get_json_body(resp).await;
+        assert_eq!(resp_body["block_timestamp"], future_time);
+
+        // create block and check if block_timestamp is less than future_time
```

Comment:
Why should it be less than future time? Didn't we update the time right away in devnet-py, as implemented here: https://github.com/0xSpaceShard/starknet-devnet/pull/416?

Reply:
"Didn't we update the time right away in devnet-py" we did and here is the same, but this test is also testing a come back to normal workflow 

- we set time to future time and we mine block
- theoretically, we can execute contracts that can be executed in the future 
- in new block we are using date time now + shift, so we come back to the default state and normal workflow (if we want to stay in the future we should use increase time), once we come back to normal workflow block_timestamp should be less than block_timestamp

Reply:
I don't understand. What is normal workflow and how are we achieving a come back to it?

Also, just now I realized I don't like this hardcoded `3376684800` for the value of `future_time`. IMO it would be better to get the current timestamp and add a large number to it. That way it will be clear that it indeed is some future moment, otherwise I have to believe you or manually check.

Reply:
I just checked, this behavior is not the same as in devnet-py. Also, compared to devnet-py, some of the responses have been changed (i.e. no `time_increased_by` property in response of `increase_time` - do we think users are not relying on it?)

Reply:
I added block_timestamp in IncreaseTimeResponse intentionally so users can have this data but I can change that to be 1:1

Reply:
5460e3a2e00d461e15ac8feb91b61378da904a39

Reply:
You didn't address the most important part of this conversation: the fact that time advancing is not done when expected. Here's how you can test it:
```rust
        // replace BackgroundDevnet process to be Devnet-py's command
        let process = Command::new("<YOUR_DEVNET_VENV>/bin/starknet-devnet")
        // let process = Command::new("cargo")
                // .arg("run")
                // .arg("--release")
                // .arg("--")
```
The test should fail at
```rust
assert!(last_block["timestamp"].as_u64() < Some(future_time));
```

Reply:
Another two things:
1. If the test name indicates it's the far future, why just set it to 100 seconds in the future?
2. #235 introduced `get_unix_timestamp_as_seconds` which can be used to obtain current timestamp.

Reply:
I added changes that we discussed on calls: b97f605deaf37e0da7d3012ad5ed40cbed25825c